### DISCUSSION TITLE
Documentation on the Auto-PID feature

### DISCUSSION THREAD

**Discussion by camueller** [meatpiHQ/wican-fw#223] [Q&A]
Is there any documentation on the Auto-PID feature?

**Comment by meatpiHQ** [comment]
@camueller its not documented yet, I'm still behind on that. What I'm trying to do now is to make it as simple as possible to connect to Home Assistant using the new integration. If you want to try it out the update to v3.46. 

I will try to get the documentation out by next week.

https://github.com/jay-oswald/ha
[wican-fw_obd_v346.zip](https://github.com/user-attachments/files/17270297/wican-fw_obd_v346.zip)
-wican

**Reply by camueller** [reply]
I'm not using Home Assistent but from what I have understood so far is that the Auto-PID feature transmits some configurable PIDs to the car right after it has become online and publishes the response messages via MQTT. Is this correct?

If so, it would allow any application to analyze those responses at a later point in time when the WiCAN is not online anymore.

**Reply by meatpiHQ** [reply]
The autopid, basically reads a PID from the car and sends the result to MQTT. For example it will automatically read the SoC and do the calculation and sends it to MQTT. At the moment it doesn't store the value it will only sends if it is able to read it.

If you use the vehicle profile all the PIDs are sent to one topic. if you're using a wallbox you can use the custom PIDs. I add a vehicle profile leaf, adding a custom PID is very similar to what's in the profile.

https://github.com/meatpiHQ/wican-fw/blob/main/vehicle_profiles/nissan/leaf-ze1.json

**Reply by camueller** [reply]
I (still) have a Leaf ZE1 and will give it a try in the next days ... Hope to get it working ...

**Reply by camueller** [reply]
I'm on FW 3.46 now.

The link you provided for Nissan Leaf profile was not correct - it is an HTML page. However I was able tro retrieve the file from the respository using this link: https://github.com/meatpiHQ/wican-fw/blob/main/vehicle_profiles/nissan/leaf-ze1.json

After I uploaded the car profile on the "Automate" tab it briefly shows "File uploaded successfully" and seem to restart - at least the browser is suddenly on the Status page. If I switch back to the "Automate" page there is not Vehicle model in the corresponding dropdow box:

![image](https://github.com/user-attachments/assets/82693b60-3d86-477f-bc95-1e01830b9059)

**Reply by meatpiHQ** [reply]
@camueller you have to upload [vehicle_profiles.json(]https://github.com/meatpiHQ/wican-fw/blob/main/vehicle_profiles.json) if you upload leaf-ze1.json it wont work.  vehicle_profiles.json contains all the supported profiles, later on WiCAN will be able to download directly from github and apply only the selected profile, but at the moment you have to upload vehicle_profiles.json

**Reply by camueller** [reply]
I did not realize that `leaf-ze1.json` was extracted from `vehicle_profiles.json`. I uploaded `vehicle_profiles.json` and was able to select the vehicle model:
![image](https://github.com/user-attachments/assets/e4795011-3772-4c93-88ad-5b45ac6685e0)

Since `vehicle_profiles.json` contains the relevant PIDs for Nissan Leaf I assume that I don't have to configure custom PIDs.
I restarted the WiCAN in order to trigger the Automate feature.
However I don't see anything published  to the Topic `wican/soc` I configured on the `Automate` tab. 

Why do it have to specifiy a cycle time? Is there any "special" value for this to indicate that it should be done once after coming online?

**Reply by meatpiHQ** [reply]
Just to confirm, did you click the store button after you selected the parameters? After reboot do you still see the configuration?

> Why do it have to specifiy a cycle time? Is there any "special" value for this to indicate that it should be done once after coming online?

Any value should be fine, this is the update rate. Keep it short for know.

**Reply by camueller** [reply]
There were not parameters presented. I simply selected the vehicle model and then "Store".
I just tried it again with same result. Still looks exactly as the screenshot I posted 12 mintes ago.

**Reply by meatpiHQ** [reply]
Thats really weird, can you see any messages on topic `wican/soc/error`?

**Reply by camueller** [reply]
No.
We could do a Google Meet session now if you are online anyway?

**Reply by meatpiHQ** [reply]
I'm online on discord. 

Many users are using this right now, not sure why its not working for you. Can you please post a screenshot of your  setting tab?

**Reply by camueller** [reply]
I tried "Custom PIDs" using the values from `vehicle_profiles.json`. I used a different topic as you may see:
![image](https://github.com/user-attachments/assets/8467b6ab-9be0-411d-a491-9108a824e3ce)

Rebooted the WiCAN again ...

Still nothing published to MQTT server except the "online" message. This means that communication toe MQTT server is working but the automate feature somehow is not.

**Comment by camueller** [comment]


**Reply by meatpiHQ** [reply]
Check your private message on discord.

You need to set the protocol to autopid, sorry forgot to mention this.
